doc/BMI_MODELS.md

```diff
@@ -259,6 +271,48 @@ Similarly, different compilers (or different compiler versions) may implement `d
 
 An example implementation for an appropriate BMI model as a **C++** shared library is provided in the project [here](../extern/test_bmi_cpp).
 
+## BMI Models Written in Python
+
+  - [Enabling Python Integration](#enabling-python-integration)
+  - [BMI Python Model as Package Class](#bmi-python-model-as-package-class)
+  - [BMI Python Example](#bmi-python-example)
+
+### Enabling Python Integration
+
+Python integration is controlled with the CMake build flag `NGEN_ACTIVATE_PYTHON`, however this currently defaults to "On"--you would need to turn this off if Python is not available in your environment. See [the Dependencies documentation](DEPENDENCIES.md#python-3-libraries) for specifics on Python requirements, but in summary you will need a working Python environment with NumPy installed. You can set up a Python environment anywhere with the usual environment variables, but note that ngen will look for a Python environment in the build directory named `.venv` or `venv` by default. The appropriate Python environment should be active in the shell when ngen is run.
+
+For Python BMI models specifically, you will also need to install the [bmipy](https://github.com/csdms/bmi-python) package, which provides a base class for Python BMI models.
+
+### BMI Python Model as Package Class
+
+To use a Python BMI model, the model needs to be installed as a package in the Python environment and the package must have a class that extends bmipy, like so
+
+```python
+from bmipy import Bmi
+class bmi_model(Bmi):
+    ...
+```
+
+**TIP:** If you are actively developing a Python BMI model, you may want to [install your package with the `-e` flag](https://pip.pypa.io/en/stable/topics/local-project-installs/#editable-installs).
+
+As noted above, Python modules require the package and class name to be specified in the realization config via the `python_class` key, such as:
+
+```javascript
+{
+  "global": {
+    "formulations": [
+        { 
+            "name": "bmi_python",
+            "params": {
+                "python_type": "mypackage.bmi_model",
+                "model_type_name": "bmi_model",
+                //...
+```
+
+### BMI Python Example
+
+An example implementation for an appropriate BMI model as a **Python** class is [provided in the project](../extern/test_bmi_py), or you can examine the CSDMS-provided [example Python model](https://github.com/csdms/bmi-example-python).
```

Comment:
Not essential, but it might be worth a quick reference to the Python BMI unit tests and their usage of this example implementation.

Reply:
I'm inclined to leave it out... I'm tempted to leave out our test class too, which implements some non-existent (deprecated?) BMI methods.